compiler/src/dotty/tools/dotc/transform/LazyVals.scala

```diff
@@ -129,50 +129,58 @@ class LazyVals extends MiniPhase with IdentityDenotTransformer {
     Thicket(field, getter)
   }
 
-  /** Replace a local lazy val inside a method,
-    * with a LazyHolder from
-    * dotty.runtime(eg dotty.runtime.LazyInt)
-    */
+  /** Desugar a local `lazy val x: Int = <RHS>` into:
+   *
+   *  ```
+   *  val x$lzy = new scala.runtime.LazyInt()
+   *
+   *  def x$lzycompute(): Int = x$lzy.synchronized {
+   *    if (x$lzy.initialized()) x$lzy.value()
+   *    else x$lzy.initialize(<RHS>)
+   *      // for a Unit-typed lazy val, this becomes `{ rhs ; x$lzy.initialize() }`
```

Comment:
It's a bit confusing to describe an optimization we don't implement, and have a TODO about that a few lines below. I would put the TODO here with the description of the optimization.